src/App.js

```diff
@@ -1,14 +1,78 @@
-import './App.css';
-import Education from './components/Education';
-import Experience from './components/Experience';
-import General from './components/General';
+import { useState } from 'react';
+import styles from './app.module.css';
+import { Education, Experience, General } from './components/';
+
+const initialData = {
+  general: { name: 'John', email: 'john@john.john', phone: '+334455667788' },
+  education: [
+    {
+      school: 'Very Nice school',
+      title: 'Pupil',
+      date: { from: '2000-06', to: '2003-07' },
+    },
+    {
+      school: 'University',
+      title: 'Student',
+      date: { from: '2004-06', to: '2009-07' },
+    },
+  ],
+  experience: [
+    {
+      company: 'BBBB',
+      position: 'Worker',
+      description: 'nice',
+      date: { from: '2010-01', to: '2015-02' },
+    },
+  ],
+};
+
 function App() {
+  const [data, setData] = useState(initialData);
```

Comment:
I would separate data to three separate state components: general, experience, education. With current setup you make your state management more complex, since you have one big object that you have to spread on change and rewrite on different actions.

Reply:
Also, now I see that you don't do anything with that `data` in <App>, so you should declare that state of each component inside that particular component.